.github/workflows/build-and-publish-image.yml

```
@@ -31,11 +35,25 @@ jobs:
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ghcr.io/${{ github.repository_owner }}/gamma
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+            type=sha
+            type=ref,event=branch
```

```suggestion
            type=edge
```
It seems like the `branch` parameter defaults to `$repo.default_branch`, so we would not need to specify `main` in this case.